### PR TITLE
CI Pin numpy 32 bit

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by topper-123 — APPROVED** [review]
LGTM. Is there a issue about this, so it can be tracked and fixed for 2.0?

**Comment by MarcoGorelli** [comment]
yup, here https://github.com/pandas-dev/pandas/issues/50392

**Comment by MarcoGorelli** [comment]
from https://github.com/pandas-dev/pandas/pull/49777 it seems there a preference for not pinning numpy unless it's necessary, so I'll just xfail the test

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

File: pandas/tests/series/test_constructors.py [hunk 1]
```diff
@@ -733,7 +736,7 @@ def test_constructor_cast(self):
         with pytest.raises(ValueError, match=msg):
             Series(["a", "b", "c"], dtype=float)
 
-    @pytest.mark.xfail(np_version_gte1p24, reason="GH49777")
+    @pytest.mark.xfail(np_version_gte1p24 and not IS64, reason="GH49777")
```
Comment (bnavigator) [inline pandas/tests/series/test_constructors.py:739]: The test also fails on 64bit

Comment (bnavigator) [reply pandas/tests/series/test_constructors.py:739]: At least for me on openSUSE. I don't know why it does XPASS on your second to last commit.

**Pull request merged**